### DISCUSSION TITLE
Additional speedtest services / tools

### DISCUSSION THREAD

**Discussion by alexjustesen** [alexjustesen/speedtest-tracker#976] [Ideas]
# Description

This discussion is pinned and tracks all currently considered additional speedtest services or tools.

## Services

- [ ] [iPerf](https://github.com/alexjustesen/speedtest-tracker/issues/64)
- [ ] [Librespeed CLI](https://github.com/librespeed/speedtest-cli)
- [ ] [Cloudflare](https://github.com/alexjustesen/speedtest-tracker/issues/284)
- [ ] [Fast](https://github.com/alexjustesen/speedtest-tracker/issues/884)
- [ ] [Open Speedtest](https://github.com/alexjustesen/speedtest-tracker/issues/1645)
- [ ] [nettest.com](https://github.com/alexjustesen/speedtest-tracker/issues/1722)

**Comment by jaxjexjox** [comment]
Please consider fast-cli

I have no programming skills at all and I've fiddled and found that it runs in Linux, it reports back results, I think in multiple formats and therefore,  you just need the code to call another engine.  I suspect the work to do this would not be super super hard, not compared to  the other work already done.

More than happy to test and assist where I can

**Comment by jaxjexjox** [comment]
You could also consider borrowing some code from here potentially:

https://github.com/vkentta/internet-speed-logger

**Reply by tbleiker** [reply]
There is also https://github.com/gnmyt/myspeed (I haven't tested it yet... still on my todo list 😄).

**Comment by jaxjexjox** [comment]
Is there anything I can do, in any capacity to help with this, without coding skills but with excellent googling skills?

Thank you again for your time and effort.

**Comment by svenvg93** [comment]
HI @alexjustesen, 
Just wondering is this is still something your considering? Normally ISPs defacto use/reqeust Ookla tests to troubleshoot speed issues. 
For the ease for mantaince having one tool might be easier to 😉

**Reply by alexjustesen** [reply]
"Considering" is the best way to put it, it's not anything I want to tackle prior to tagging v1.

**Reply by svenvg93** [reply]
IMHO adding more tools might make it more complicated to have it all working correctly. And like I mentioned is that ISP mostly only trust Ookla for tests like this. 

Adding a ping based monitoring as mentioned here https://github.com/alexjustesen/speedtest-tracker/issues/826 might be a good addition to the speedtest.

**Reply by jaxjexjox** [reply]
> IMHO adding more tools might make it more complicated to have it all working correctly. And like I mentioned is that ISP mostly only trust Ookla for tests like this.
> 
> Adding a ping based monitoring as mentioned here #826 might be a good addition to the speedtest.

The focus of this was about opening other tools, hence me first asking for it.
Speedtest-cli simply won't work for me.

Most of these tools already include a ping.

I just need to use fast-cli and to my knowledge, there's already multiple docker containers which take care of fast-cli, plus if I recall the output content is somewhat similar to the speedtest-cli output.

Here you can see someone attempting to help me, sadly lacking in skills in attempting the same thing for a different docker based speedtest application.

https://github.com/ansemjo/speedtest-plotter/issues/154#issuecomment-1784671234

**Reply by svenvg93** [reply]
> 
> The focus of this was about opening other tools, hence me first asking for it. Speedtest-cli simply won't work for me.
> 

I Do understand the focus, im just trying to make sure there won't be a wild amount of diffrent tools integrated as it makes maintaining it complicated. Especially as we are all relying on Alex for it ;)

Out of curiosity why does the Speedtest CLI not work for you? Maybe we can help with that. 

> 
> Most of these tools already include a ping.
> 

Im didn't mean just a ping during the speedtest, but more a smokeping like integration as it will add the ability to monitor latency to endpoints. Together with a Speedtest (Whichever one it might be ;)) it will give a powerfull and easy tool to monitor your internet.

Just be clear; Im not against adding diffrent tools, just giving my personal and professional opinion.

**Reply by jaxjexjox** [reply]
> > The focus of this was about opening other tools, hence me first asking for it. Speedtest-cli simply won't work for me.
> 
> I Do understand the focus, im just trying to make sure there won't be a wild amount of diffrent tools integrated as it makes maintaining it complicated. Especially as we are all relying on Alex for it ;)
> 
> Out of curiosity why does the Speedtest CLI not work for you? Maybe we can help with that.
> 
> > Most of these tools already include a ping.
> 
> Im didn't mean just a ping during the speedtest, but more a smokeping like integration as it will add the ability to monitor latency to endpoints. Together with a Speedtest (Whichever one it might be ;)) it will give a powerfull and easy tool to monitor your internet.
> 
> Just be clear; Im not against adding diffrent tools, just giving my personal and professional opinion.


This is due to the network I'm on and the devices being trusted / untrusted.   The ZScaler system is refusing traffic from my machine running speedtest-tracker, due to the port speedtest-cli runs on.  
This is unfortunately to be expected, I'm fine with that.

Luckily, fast-cli runs on an allowed port and it the results are in a similar layout to speedtest-cli.   If I had a bit more programmer chops, I'd do it myself.

https://github.com/ansemjo/speedtest-plotter/issues/154#issuecomment-1784671234

**Comment by jaxjexjox** [comment]
Hey Alex, is there anything I can do, with little to no skills to help?   
I google this every couple of months and not a single, decent tool exists which doesn't use speedtest.
This is the ONLY tool I've found even close:

https://github.com/vkentta/internet-speed-logger

**Reply by svenvg93** [reply]
Think most of them use Ookla as it is the standard for ISPs to trust compared to the other tools. 
You can always try to add support for other tools (with some help from google and ai maybe ;)) by following the steps here 
https://docs.speedtest-tracker.dev/contributing/development-environment

**Comment by vinii8817** [comment]
Consider include OpenSpeedTest, as this would allow tests in private networks. For example, I would like to test the performance between my server in my homelab and the server I have on DigitalOcean. I would have an OpenSpeedTest server on DigitalOcean and the Tracker in my homelab. 

This would make it more flexible since I would have complete control over both ends of the test, ensuring the performance between the two locations.

I believe this could also help others replace iPerf. With the ease of deployment of both OpenSpeedTest and the Tracker, it would be a very useful and simple pair.

**Reply by jaxjexjox** [reply]
> Consider include OpenSpeedTest, as this would allow tests in private networks. For example, I would like to test the performance between my server in my homelab and the server I have on DigitalOcean. I would have an OpenSpeedTest server on DigitalOcean and the Tracker in my homelab.
> 
> This would make it more flexible since I would have complete control over both ends of the test, ensuring the performance between the two locations.
> 
> I believe this could also help others replace iPerf. With the ease of deployment of both OpenSpeedTest and the Tracker, it would be a very useful and simple pair.


OpenSpeedtest definitely has use and I use it at home, however it already has a great existing docker container, just no graphing / history.
https://hub.docker.com/r/openspeedtest/latest

I personally have use of a tool myself where I have multiple machines talking to each other across my WAN and being able to identify graph and track it.

That being said, this adds a fair bit of complexity.

**Reply by vinii8817** [reply]
I understand, I also have some graphs, but I would like to periodically test the possible bandwidth limit. Sometimes, even internally, when testing a firewall or switch, I find it would be useful. The container works very well; I use it. 

The intention of including OpenSpeedTest is to provide the possibility of testing in controlled environments, not just for the internet.

**Reply by tbleiker** [reply]
I would love to test against a self-hosted server (OpenSpeedTest, librespeed) as well in order to monitor speed within my own network or with an own external server (e.g. for off-site backup).

**Reply by Druidblack** [reply]
Is there any information about adding OpenSpeedTest?

**Reply by arvkonstantin** [reply]
Upvote OpenSpeedTest

**Comment by KristofKekesi** [comment]
Do we have an ETA on the Cloudflare speedtest?

**Comment by arvkonstantin** [comment]
Ookla servers are not available in Russia, so it would be great to be able to use other services. I have deployed a speedtest in my home lab, but I cannot use it. I will wait for alternatives to Ookla.
